continuous_integration/recipes/dask/meta.yaml

```diff
@@ -13,16 +12,16 @@ source:
 
 build:
   number: {{ GIT_DESCRIBE_NUMBER }}
-  string: py{{ py_version }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
+  string: py_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
   noarch: python
 
 requirements:
   host:
     - python >=3.7
   run:
     - python >=3.7
-    - dask-core {{ dask_version }}={{ dask_build }}
-    - distributed {{ version }}=py{{ py_version }}_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
+    - dask-core >={{ dask_version }}={{ dask_build }}
+    - distributed {{ version }}=*_{{ GIT_DESCRIBE_HASH }}_{{ GIT_DESCRIBE_NUMBER }}
```

Comment:
Do we still want to keep `distributed` as tightly pinned?

Reply:
After some more thought here, I think the better solution here would be to build Dask / Distributed nightlies on commits to Dask as well - opened up https://github.com/dask/dask/pull/8654 doing that.

With that in, we should be safe to keep `dask-core` tightly pinned here

Reply:
What happens if there is a change in Distributed without a corresponding change in Dask?

Thinking about this a bit more I wonder if there is a lighter weight solution. Could the Dask Conda GH Action just trigger the GH Action on Distributed?

Reply:
> What happens if there is a change in Distributed without a corresponding change in Dask?

In that case, the workflow in this repo would build new `distributed` / `dask` pre-releases pinned to the most recent `dask-core` pre-release, which is what we want. As long as we're triggering `distributed` / `dask` builds on commits to either repo we should be good.

> Could the Dask Conda GH Action just trigger the GH Action on Distributed?

That sounds like an interesting solution - I'll look into that now, though I'd imagine to do it we would need to generate a [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token) to send the GHA request across repos

Reply:
Ah I see. So both repos will be building and uploading the same packages `dask-core`, `dask`, and `distributed`. Is that right? The only potential issue here is if both upload packages at the same time. While rare, this can cause the same package to appear twice on the channel making neither of them downloadable.

Thanks Charles!

Reply:
Close - in Dask's case, all 3 would be built. In the case of Distributed only `distributed` and `dask` would be built, pinned to the most recent version of `dask-core`.

> While rare, this can cause the same package to appear twice on the channel

I considered this, though I was expecting identical packages to overwrite one another rather than coexist. In that case the better option would probably be to have the Dask workflow trigger builds in Distributed, and use GHA's [concurrency control](https://docs.github.com/en/actions/using-jobs/using-concurrency) to ensure that only one conda build / upload is running per Git ref / hash

Reply:
I did some investigation on this and attempted to upload a duplicate package to my personal channel:

```
anaconda upload dask-core-2022.01.1-py310_g6824e5dd_13.tar.bz2
Using Anaconda API: https://api.anaconda.org
[WARNING] Token warning: expires in 0 days
Using "charlesbluca" as upload username
Processing 'dask-core-2022.01.1-py310_g6824e5dd_13.tar.bz2'
Detecting file type...
File type is "conda"
Extracting conda package attributes for upload
Creating package "dask-core"
Creating release "2022.01.1"
Uploading file "charlesbluca/dask-core/2022.01.1/noarch/dask-core-2022.01.1-py310_g6824e5dd_13.tar.bz2"
Distribution already exists. Please use the -i/--interactive or --force or --skip options or `anaconda remove charlesbluca/dask-core/2022.01.1/noarch/dask-core-2022.01.1-py310_g6824e5dd_13.tar.bz2
[ERROR] ('file noarch/dask-core-2022.01.1-py310_g6824e5dd_13.tar.bz2 already exists for package dask-core version 2022.01.1', 409)
```

Does this hold true for organizational Anaconda channels? If so then we should be safe to build all 3 packages from both repos, as long as we think carefully about when / where to use the `--force` flag to force an overwrite of the currently published package(s).

I _think_ the only place we would need `--force` is in builds triggered by Dask, as the only time we would need to overwrite something is to replace the pinning for a `dask` / `distributed` pre-release, and that should only happen if we commit to Dask some time after a commit to Distributed.

> So both repos will be building and uploading the same packages `dask-core`, `dask`, and `distributed`.

After thinking about this more, I do think that building / uploading all 3 packages on commits to either repo is the way to go, as it generally minimizes the lag between a Dask commit being pushed and Distributed being made aware of a new `dask-core` pre-release to pin to.

Also I realize that my explanations may not be the greatest here - happy to chat about this offline 🙂

Reply:
Yeah this issue definitely comes up. It has bitten RAPIDS and conda-forge a few times. It's not easy to reproduce, the fix is relatively trivial (delete one package), but is preferable to avoid since that usually involves tracking someone down with permissions to fix.

`--force` wouldn't help as Anaconda will upload both without `--force` because it doesn't think either one is already there. It is (I'm guessing) effectively a concurrency bug. IOW both uploads check with the server for a package, neither one finds a package, both uploads succeed. Users try to install and get checksum mismatch errors.

Happy to discuss more offline 🙂